src/test/java/com/google/collegeplanner/TestCourseList.java

```diff
@@ -36,9 +36,44 @@ public void servletResponseHasCourses() throws Exception {
     // Tests that courses_detailed exists
     Assert.assertNotNull(coursesDetailed);
     // Checks that the correct number of JSON Objects are contained
-    Assert.assertEquals(coursesDetailed.size(), 18);
+    Assert.assertEquals(coursesDetailed.size(), 30);
     // Checks whether the first one is CMSC101
     String firstCourse = coursesDetailed.get(0).toString();
-    Assert.assertEquals(firstCourse, "{\"name\":\"CMSC101\"}");
+    String expectedJson = "{"
+        + "\"course_id\":\"AASP100\","
+        + "\"core\":[\"SH\",\"D\"],"
+        + "\"relationships\":{"
+        + "\"coreqs\":null,"
```

Comment:
can you indent here so it's slightly easier for humans to see the structure?

``
+ "\"relationships\":{"
+ "  \"coreqs\": null,"
```

Reply:
I did indent everything properly but the auto formatter changes it to this.

Reply:
wait, the formatter removed whitespace inside of the string? That doesn't seem right. I could see that it would remove whitespace outside the string.

Reply:
If he changes the string itself, will it still match with the JSON string retrieved from the servlet?

Reply:
Oh sorry, I saw that wrong. I added tabs to format the code, not inside the strings. But anyways, adding spaces to the string would cause the equals-assertion to fail, right?

Reply:
Is there an easy "JsonEquals" assertion we can use that is whitespace/formatting-agnostic?

Reply:
If so, we could do:

assertJsonEquals(coursesDetailed.get(0), Json.parse(expectedJson));

Reply:
Found a third-party library to do this: https://github.com/skyscreamer/JSONassert